account_payment_order/report/account_payment_order.xml

```diff
@@ -76,7 +76,8 @@
                                 <span t-esc="get_bank_account_name(line.partner_bank_id)"/>
                             </td>
                             <td class="text-center">
-                                <span t-raw="'%s &lt;br&gt;' % line.move_line_id.move_id.name if line.move_line_id.move_id else ''"/>
+                                <span t-raw="'%s &lt;br&gt;' % line.move_line_id.invoice_id.number" t-if="line.move_line_id.invoice_id"/>
+                                <span t-raw="'%s &lt;br&gt;' % line.move_line_id.move_id.name if line.move_line_id.move_id else ''" t-if="not line.move_line_id.invoice_id"/>
```

Comment:
`line.move_line_id.invoice_id.number or line.move_line_id.move_id.name or ''` should work too and is maybe slightly easier to read.